### PR TITLE
Addresses image stretching: #4492

### LINKED ISSUE #4492: Image covers should never appear stretched.
Book covers sometimes look stretched
e.g.
Winnie the Pooh on https://openlibrary.org/subjects/in_library#ebooks=true&sort=edition_count
Screen Shot 2021-01-25 at 8 54 52 PM

CSS has an `object-fit: contains` rule which will make this long much nicer:
Screen Shot 2021-01-25 at 8 55 22 PM

Let's use it!

Note:  We should review all pages which display covers, and apply it to those too, not just the subjects page. 

When testing at least verify the fix works on:
* List pages
* Home page
* Subjects page

### Stakeholders
@jdlrobson @mekarpeles

### PR DESCRIPTION
**Pull request by mitchell-frost**
Closes #4492 

### Testing
All the tests work. The only problem is that the local version I have built with docker does not seem to have "Winnie_the_Pooh". Hence, I added the object-fit wherever I saw an image.

### Stakeholders
@jdlrobson

### CODE REVIEW

**Review by Yashs911 — CHANGES_REQUESTED** [review]
@mitchell-frost Thanks for the PR. But I don't think this is the right solution.
image
You need to apply object-fit to all the book covers
and you also need to add background color according to https://github.com/internetarchive/openlibrary/issues/4492#issuecomment-771872381

For importing a book:
1) `docker-compose exec web bash`
2) `./scripts/copydocs.py /books/OL22906339M`

you can replace` /books/OL22906339M` with other terms such as` /works/OLxxxxxxx`

File: static/css/components/carousel.less [hunk 1]
```diff
@@ -18,13 +18,15 @@
 .carousel-container {
   margin: 0 -20px;
   padding: 3px 15px;
+  object-fit: contain;
```
Comment (Yashs911) [inline static/css/components/carousel.less:21]: Not required here

File: static/css/components/carousel.less [hunk 2]
```diff
@@ -18,13 +18,15 @@
...
 }
 
 .carousel {
   .display-flex();
   margin-bottom: 15px;
   overflow-x: scroll;
   padding: 10px 20px;
+  object-fit: contain;
```
Comment (Yashs911) [inline static/css/components/carousel.less:29]: Not required here

File: static/css/components/listLists.less [hunk 3]
```diff
@@ -26,6 +26,7 @@ ul.listLists {
       width: 42px;
       min-height: 35px;
       padding: 0 10px;
+      object-fit: contain;
       /* stylelint-disable max-nesting-depth */
       img {
```
Comment (Yashs911) [inline static/css/components/listLists.less:31]: Should be applied here. + Background color

File: static/css/components/lists-page-cta.less [hunk 4]
```diff
@@ -25,5 +25,6 @@
     background-position: 35px 35px;
     padding-left: 130px;
     padding-right: 60px;
+    object-fit: contain;
```
Comment (Yashs911) [inline static/css/components/lists-page-cta.less:28]: Not required here.

File: static/css/components/mybooks.less [hunk 5]
```diff
@@ -9,6 +9,7 @@
 .mybooks {
   border-top: 1px solid @lightest-grey;
   margin-top: 10px;
+  object-fit: contain;
```
Comment (Yashs911) [inline static/css/components/mybooks.less:12]: Not required here.

**Comment by Yashs911** [comment]
@mitchell-frost You need to apply the object-fit property on:

1. homepage/subject carousel book covers.
2. lists covers https://openlibrary.org/lists
3. subject list cover https://openlibrary.org/subjects/architecture 
image

**Comment by mitchell-frost** [comment]
@Yashs911 thanks a lot! The code base is a bit overwhelming for me. Any help on where I can find the CSS for these specific pages?

**Comment by Yashs911** [comment]
> @Yashs911 thanks a lot! The code base is a bit overwhelming for me. Any help on where I can find the CSS for these specific pages?

@mitchell-frost what I generally do is:
Inspect the element and then search that CSS selector using a code editor (VSCode).
For e.g. homepage carousel book cover css selctor is `.book img.bookcover`
On searching for .bookcover you will  find the CSS in `static\css\components\book.less` (https://github.com/internetarchive/openlibrary/blob/master/static/css/components/book.less)

**Review by jdlrobson — CHANGES_REQUESTED** [review]
Per @Yashs911

**Comment by Yashs911** [comment]
@mitchell-frost Any update on this.

**Comment by jdlrobson** [comment]
Closing this due to inactivity to free up the task for another contributor. Feel free to reopen at any time. Thank you for your contribution!

**Pull request closed without merging**